src/main/java/org/opensearch/plugin/insights/core/service/categorizer/QueryShapeGenerator.java

```diff
@@ -243,18 +198,20 @@ static String buildSortShape(List<SortBuilder<?>> sortBuilderList, Boolean showF
      * @return String: comma separated list with leading space in square brackets
      * Ex: " [my_field, width:5]"
      */
-    static String buildFieldDataString(List<Function<Object, String>> methods, NamedWriteable builder) {
+    static String buildFieldDataString(Map<Class<?>, List<Function<Object, String>>> fieldMethodsMap, NamedWriteable builder) {
```

Comment:
`fieldMethodsMap` paramater can be removed now since we have the same instance across query, sort and aggregation invocations